src/WholeBodyControllers/src/IntegrationBasedIK.cpp

```diff
@@ -0,0 +1,229 @@
+/**
+ * @file IntegrationBasedIK.h
+ * @authors Giulio Romualdi
+ * @copyright 2022 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ */
+
+#include <BipedalLocomotion/Conversions/ManifConversions.h>
+#include <BipedalLocomotion/IK/JointTrackingTask.h>
+#include <BipedalLocomotion/IK/R3Task.h>
+#include <BipedalLocomotion/IK/SE3Task.h>
+#include <BipedalLocomotion/IK/SO3Task.h>
+#include <BipedalLocomotion/TextLogging/Logger.h>
+
+#include <WalkingControllers/WholeBodyControllers/IntegrationBasedIK.h>
+
+#include <iDynTree/Core/EigenHelpers.h>
+#include <iDynTree/Core/Position.h>
+#include <iDynTree/Core/Twist.h>
+#include <iDynTree/Core/VectorFixSize.h>
+#include <iDynTree/KinDynComputations.h>
+
+#include <manif/manif.h>
+
+#include <memory>
+
+using namespace WalkingControllers;
+
+bool IntegrationBasedIK::initialize(
+    std::weak_ptr<const BipedalLocomotion::ParametersHandler::IParametersHandler> handler,
+    std::shared_ptr<iDynTree::KinDynComputations> kinDyn)
+{
+    constexpr auto prefix = "[IntegrationBasedIK::initialize]";
+    constexpr std::size_t highPriority = 0;
+    constexpr std::size_t lowPriority = 1;
+
+    auto ptr = handler.lock();
+    if (ptr == nullptr)
+    {
+        BipedalLocomotion::log()->error("{} Invalid parameter handler", prefix);
+        return false;
+    }
+
+    m_usejointRetargeting = false;
+    int usejointRetargeting = 0;
+    ptr->getParameter("use_joint_retargeting", usejointRetargeting);
+    m_usejointRetargeting = usejointRetargeting != 0;
```

Comment:
`use_joint_retargeting` is actually a number stored in the main configuration file. 
```cpp
ptr->getParameter("use_joint_retargeting", m_usejointRetargeting);
```
will fail since `m_usejointRetargeting` is a bool and the check inside blf fails. We should change the code in blf

Reply:
By why this works instead? https://github.com/robotology/walking-controllers/blob/20b1e8f8f7d8daa806d282b14eb9c4bf89275a5f/src/WholeBodyControllers/src/IntegrationBasedIK.cpp#L52-L53

Reply:
Because the parameter `use_root_link_for_height` is a boolean. By the way, this https://github.com/ami-iit/bipedal-locomotion-framework/pull/566 should remove this limitation of blf

Reply:
Fixed here: https://github.com/robotology/walking-controllers/pull/118/commits/97b6c70ca4ded0f09e979dcfab5651e7e9eef9e3